names_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fake_test
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/cilium/fake"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNames(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Error("Names(-1) should panic()")
+		}
+	}()
+	_ = fake.Names(-1)
+
+	names := fake.Names(0)
+	assert.Nil(t, names)
+
+	max := 100
+	names = fake.Names(rand.Intn(max + 1))
+	assert.NotNil(t, names)
```

Comment:
I think this test may fail:

1. When the test pick `max=0`
2. When `Names` pick `n=0`

Both cases because `rand.Intn` may always return `0`. `names = fake.Names(rand.Intn(max) + 1)` would solve 1. but not 2. so I think we should either remove the `NotNil` check here, or make `Names` return an non-`nil` empty slice when `n=0`.